File: Python/Tree/Queue.py
print('Imported Sucessfully')
class Queue:
    def __init__(self):
        self.list1 = []
        self.front = 0
        self.rear  = 0
    
    def Create(self):
        self.front = 0
        self.rear = -1

    def Enqueue(self,element):
            self.list1.append(element)
            self.rear = self.rear+1
            
    def Dequeue(self):
        x=0
        if (self.front==self.rear):print('Queue is empty')
        else:
            x = self.list1[self.front]
            self.list1.remove(self.list1[self.front])
            self.rear = self.rear-1
            return x
        
    def isEmpty(self):
        if self.rear==self.front:return 1
        else: return 0
    
    def Display(self):
        print(*self.list1)

File: C++/Tree/Creation of tree/Queue.c
                          // CIRCULAR QUEUE
struct Queue
{
  int size;
  int front;
  int rear;
  struct Node **Q; // int *Q  Since it already a pointer
                   // struct Node* => For Storing pointer of type Node , *Q => array,
};
    
void Create_Queue(struct Queue *q,int size)
        {
          q->size=size;
          q->front=q->rear=0;
          q->Q=(struct Node**)malloc(q->size*sizeof(struct Node *)); // Array of pointers, Should type cast this as node type of pointer
        }

void enqueue(struct Queue *q,struct Node* x)
        {
           if((q->rear+1)%q->size==q->front)
           printf("Queue is Full");
          else
             {
              q->rear=(q->rear+1)%q->size;
              q->Q[q->rear]=x;
              }
        }

struct Node* dequeue(struct Queue *q)
   		{
		    struct Node* x=NULL;
		    if(q->front==q->rear)
		     {
		     	printf("Queue is Empty\n");
		     }
	     	else
	        {
	         q->front=(q->front+1)%q->size;
	         x=q->Q[q->front];
	        }
	    	return x;
   		}

int isEmpty(struct Queue q)
  		{
      		return q.front==q.rear; // Returns 1 if it is TRUE
  		}

/*##########################End of queue #######################*/